src/main/kotlin/org/jitsi/nlj/rtp/bandwidthestimation/GoogleCcEstimator.kt

```diff
@@ -0,0 +1,107 @@
+package org.jitsi.nlj.rtp.bandwidthestimation
+
+import java.time.Duration
+import java.time.Instant
+import kotlin.properties.Delegates
+import org.jitsi.nlj.stats.NodeStatsBlock
+import org.jitsi.nlj.util.Bandwidth
+import org.jitsi.nlj.util.DataSize
+import org.jitsi.nlj.util.bps
+import org.jitsi.nlj.util.createChildLogger
+import org.jitsi.nlj.util.kbps
+import org.jitsi.nlj.util.mbps
+import org.jitsi.utils.logging.DiagnosticContext
+import org.jitsi.utils.logging2.Logger
+import org.jitsi_modified.impl.neomedia.rtp.remotebitrateestimator.RemoteBitrateEstimatorAbsSendTime
+import org.jitsi_modified.impl.neomedia.rtp.sendsidebandwidthestimation.SendSideBandwidthEstimation
+
+private val defaultInitBw: Bandwidth = 2.5.mbps
+private val defaultMinBw: Bandwidth = 30.kbps
+private val defaultMaxBw: Bandwidth = 20.mbps
+
+class GoogleCcEstimator(diagnosticContext: DiagnosticContext, parentLogger: Logger) :
+    BandwidthEstimator(diagnosticContext) {
+    override val algorithmName = "Google CC"
+
+    /* TODO: Use configuration service to set this default value. */
+    override var initBw: Bandwidth = defaultInitBw
+    /* TODO: observable which sets the components' values if we're in initial state. */
+
+    override var minBw: Bandwidth by Delegates.observable(defaultMinBw) {
+        _, _, newValue ->
+        bitrateEstimatorAbsSendTime.setMinBitrate(newValue.bps.toInt())
+        sendSideBandwidthEstimation.setMinMaxBitrate(newValue.bps.toInt(), maxBw.bps.toInt())
+    }
+
+    override var maxBw: Bandwidth by Delegates.observable(defaultMaxBw) {
+        _, _, newValue ->
+        sendSideBandwidthEstimation.setMinMaxBitrate(minBw.bps.toInt(), newValue.bps.toInt())
+    }
+
+    private val logger = parentLogger.createChildLogger(GoogleCcEstimator::class)
+
+    /**
+     * Implements the delay-based part of Google CC.
+     */
+    private val bitrateEstimatorAbsSendTime = RemoteBitrateEstimatorAbsSendTime(diagnosticContext, logger).also {
+        it.setMinBitrate(minBw.bps.toInt())
+    }
+
+    /**
+     * Implements the loss-based part of Google CC.
+     */
+    private val sendSideBandwidthEstimation = SendSideBandwidthEstimation(diagnosticContext, initBw.bps.toLong(), logger).also {
+        it.setMinMaxBitrate(minBw.bps.toInt(), maxBw.bps.toInt())
+    }
+
+    override fun doProcessPacketArrival(now: Instant, sendTime: Instant?, recvTime: Instant?, seq: Int, size: DataSize, ecn: Byte) {
+        if (sendTime != null && recvTime != null) {
+            bitrateEstimatorAbsSendTime.incomingPacketInfo(now.toEpochMilli(),
+                    recvTime.toEpochMilli(), sendTime.toEpochMilli(), size.bytes.toInt())
+        }
+        sendSideBandwidthEstimation.updateReceiverEstimate(bitrateEstimatorAbsSendTime.latestEstimate)
+        sendSideBandwidthEstimation.updateReceiverBlock(0 /* fraction_lost */, 1 /* number_of_packets */, now.toEpochMilli())
```

Comment:
i'm wondering if it makes sense for `sendSideBandwidthEstimation` to expose a more high-level api (`processPacketArrival` or something) and hide this `updateReceiverBlock`  logic as an implementation detail  (same for `doProcessPacketLoss` below).  it just feels a bit weird to have this class hard-code these values, and i'm wondering if those make more sense as implementation constants inside of `sendSideBandwidthEstimation`?

Reply:
Yeah, this is a bit of a hack to adapt its old behavior (consuming RTCP receiver reports) to the new `BandwidthEstimator` abstraction.  I'll think about if there's a cleaner way to do this.

Reply:
new way lgtm